Source/WebCore/css/CSSCounterValue.cpp

```diff
@@ -28,32 +28,33 @@
 
 #include "CSSMarkup.h"
 #include "CSSValueKeywords.h"
+#include <wtf/PointerComparison.h> 
 #include <wtf/text/StringBuilder.h>
 
 namespace WebCore {
 
-CSSCounterValue::CSSCounterValue(AtomString identifier, AtomString separator, CSSValueID listStyle)
+CSSCounterValue::CSSCounterValue(AtomString identifier, AtomString separator, RefPtr<CSSValue> listStyleType)
     : CSSValue(CounterClass)
     , m_identifier(WTFMove(identifier))
     , m_separator(WTFMove(separator))
-    , m_listStyle(listStyle)
+    , m_listStyleType(WTFMove(listStyleType))
 {
 }
 
-Ref<CSSCounterValue> CSSCounterValue::create(AtomString identifier, AtomString separator, CSSValueID listStyle)
+Ref<CSSCounterValue> CSSCounterValue::create(AtomString identifier, AtomString separator, RefPtr<CSSValue> listStyleType)
 {
-    return adoptRef(*new CSSCounterValue(WTFMove(identifier), WTFMove(separator), listStyle));
+    return adoptRef(*new CSSCounterValue(WTFMove(identifier), WTFMove(separator), WTFMove(listStyleType)));
 }
 
 bool CSSCounterValue::equals(const CSSCounterValue& other) const
 {
-    return m_identifier == other.m_identifier && m_separator == other.m_separator && m_listStyle == other.m_listStyle;
+    return m_identifier == other.m_identifier && m_separator == other.m_separator && arePointingToEqualData(m_listStyleType, other.m_listStyleType);
 }
 
 String CSSCounterValue::customCSSText() const
 {
-    auto listStyleSeparator = m_listStyle == CSSValueDecimal ? ""_s : ", "_s;
-    auto listStyleLiteral = m_listStyle == CSSValueDecimal ? ""_s : nameLiteral(m_listStyle);
+    auto listStyleSeparator = m_listStyleType->valueID() == CSSValueDecimal ? ""_s : ", "_s;
+    auto listStyleLiteral = m_listStyleType->valueID() == CSSValueDecimal ? ""_s : nameLiteral(m_listStyleType->valueID());
```

Comment:
Is valueID() correct for counter-styles that are not in the CSSValueKeywords list?

Reply:
using `isValueID()` is fine because it returns CSSValueInvalid when the CSSValue is not a ValueID, however I need to transform the lambda I created for converting to string into a function so I can re-use it here.

Reply:
Is `CSSValueInvalid` desired? If we have a non-UA counter style name, it should be reflected.